spanner/api/Spanner/Program.cs

```diff
@@ -1343,6 +1378,177 @@ await connection.CreateDdlCommand($"DROP TABLE {table}")
+                    var partitions = await cmd.GetReaderPartitionsAsync();
+                    var transactionId = transaction.TransactionId;
+                    await Task.WhenAll(partitions.Select(
+                            x => DistributedReadWorkerAsync(x, transactionId)))
+                                .ConfigureAwait(false);
+                }
+                Console.WriteLine($"Done reading!  Total rows read: "
+                    + $"{s_rowsRead:N0} with {s_partitionId} partition(s)");
+            }
+        }
+
+        private static int s_partitionId;
+        private static int s_rowsRead;
+
+        private static async Task DistributedReadWorkerAsync(
+            CommandPartition readPartition, TransactionId id)
+        {
+            var localId = Interlocked.Increment(ref s_partitionId);
+            using (var connection = new SpannerConnection(id.ConnectionString))
+            using (var transaction = connection.BeginReadOnlyTransaction(id))
+            {
+                using (var cmd = connection.CreateCommandWithPartition(
+                    readPartition, transaction))
+                {
+                    using (var reader =
+                        await cmd.ExecuteReaderAsync().ConfigureAwait(false))
+                    {
+                        while (await reader.ReadAsync())
+                        {
+                            Interlocked.Increment(ref s_rowsRead);
+                            Console.WriteLine($"Partition ({localId}) "
+                                + $"{reader.GetFieldValue<string>("SingerId")}"
+                                + $" {reader.GetFieldValue<string>("FirstName")}"
+                                + $" {reader.GetFieldValue<string>("LastName")}");
+                        }
+                    }
+                }
+                Console.WriteLine($"Done with single reader {localId}.");
+            }
+        }
+        // [END spanner_batch_client]
+
+        public static object DeleteDatabase(string projectId,
+            string instanceId, string databaseId)
+        {
+            var response =
+                DeleteDatabaseAsync(projectId, instanceId, databaseId);
+            Console.WriteLine("Waiting for operation to complete...");
+            response.Wait();
+            Console.WriteLine($"Operation status: {response.Status}");
+            Console.WriteLine($"Deleted sample database {databaseId} on "
+                + $"instance {instanceId}");
+            return ExitCode.Success;
+        }
+
+        private static async Task DeleteDatabaseAsync(string projectId,
+            string instanceId, string databaseId)
+        {
+            string AdminConnectionString = $"Data Source=projects/{projectId}/"
+                + $"instances/{instanceId}";
+            using (var connection = new SpannerConnection(AdminConnectionString))
+            using (var cmd = connection.CreateDdlCommand(
+                $@"DROP DATABASE {databaseId}"))
```

Comment:
You should document the assumptions of DDL command. Typically you need extra permissions besides R/W on the DB to delete it.

Reply:
Added a comment to note the IAM Database Admin role is required.